run_tests.py

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env python
+
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import os
+import signal
+from subprocess import check_call, Popen, check_output
+import sys
+import time
+import pip
+import urllib2
+import tarfile
+import shutil
+
+
+DIR_TEST_ENV = "test/env"
+DIR_JENKINS_ENV = 'jenkins-env'
+VERSION_VIRTUALENV = '1.9.1'
+
+os.environ["JENKINS_HOME"] = "jenkins-master"
+
+
+def python(*args):
+    pypath = [os.path.join(DIR_TEST_ENV, 'bin', 'python')]
+    check_call(pypath + list(args))
+
+
+def pip(*args):
+    pippath = [os.path.join(DIR_TEST_ENV, 'bin', 'pip')]
+    check_call(pippath + list(args))
+
+
+def download(url, destination):
+    response = urllib2.urlopen(url)
+    with open(destination, "wb") as code:
+        code.write(response.read())
+
+
+def kill_jenkins():
+    """Kill all Jenkins processes"""
+    print "Killing Jenkins"
+    pids = check_output(['lsof', '-i:8080', '-t'])
+    pids = pids.strip().split('\n')
+
+    for pid in pids:
+        os.kill(int(pid), signal.SIGKILL)
+
+
+def run_tests():
+    """Main Function that handles all the tests."""
+    try:
+        print "Checking Patches"
+        check_call('./test/check_patches.sh')
```

Comment:
Please reduce the size of code which has been put into this try/except. The message you show in except only relates to activating the environment. So if necessary use multiple try/except or finally blocks.

This specific `check_call` here can perfectly be moved out of the try block. It throws an exception, but there is nothing we would have to clean-up.